src/interface/blas2_interface.hpp

```diff
@@ -119,16 +123,26 @@ typename sb_handle_t::event_t _gemv_impl(sb_handle_t& sb_handle, index_t _M,
       auto assignOp = make_op<Assign>(vy, addOp);
 
       // exectutes the above expression tree to yield the final GEMV result
-      return concatenate_vectors(gemvEvent,
-                                 sb_handle.execute(assignOp, local_range));
+      ret = concatenate_vectors(
+          gemvEvent, sb_handle.execute(assignOp, local_range, gemvEvent));
+
     } else {
       auto alphaMulDotsOp =
           make_op<ScalarOp, ProductOperator>(_alpha, dot_products_matrix);
       auto assignOp = make_op<Assign>(vy, alphaMulDotsOp);
-      return concatenate_vectors(gemvEvent,
-                                 sb_handle.execute(assignOp, local_range));
+      ret = concatenate_vectors(
+          gemvEvent, sb_handle.execute(assignOp, local_range, gemvEvent));
     }
 
+#ifdef DEFAULT_CPU
+    auto free_dt_buffer =
+        blas::helper::enqueue_deallocate < is_usm
+            ? helper::AllocType::usm
+            : helper::AllocType::buffer >
+                  (ret, dot_products_buffer, sb_handle.get_queue());
+    ret =
+        concatenate_vectors(ret, typename sb_handle_t::event_t{free_dt_buffer});
```

Comment:
In order to remove the error due to queue profiling that you mentioned [here](https://github.com/codeplaysoftware/sycl-blas/blob/60a808d30b33dbd0efe4a9fcbe6d07396c98c1a0/include/sycl_blas_helper.h#L86), you can remove  `free_dt_buffer` from the event list that you return.

The SYCL [specification](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:interface.queue.class) guarantee us that the corresponding operation will be executed in any case.
```
A SYCL queue may be destroyed even when there are uncompleted commands that have been submitted to the queue. Doing so does not block. Instead, any commands that have been submitted to the queue begin execution when their requisites are satisfied, just as they would had the queue not been destroyed. Any event objects for those commands are signaled in the normal manner when the command completes. Resources associated with the queue will be freed by the time the last command completes.
```

Reply:
This change has been addressed in https://github.com/codeplaysoftware/sycl-blas/pull/436